### PR TITLE
[Merged by Bors] - chore(Topology/LocallyClosed): move around

### PR DESCRIPTION
**Pull request by urkud**
- move definitions to `Topology/Defs/Basic`;
- move the last lemma to `Topology/LocalAtTarget`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Defs/Basic.lean [hunk 1]
```diff
@@ -98,6 +98,12 @@ class IsClosed (s : Set X) : Prop where
 def IsClopen (s : Set X) : Prop :=
   IsClosed s ∧ IsOpen s
 
+/--
+A set is locally closed if it is the intersection of some open set and some closed set.
+Also see `isLocallyClosed_tfae`.
```
Comment (erdOne) [inline Mathlib/Topology/Defs/Basic.lean:103]: Can you mention the file `Mathlib/Topology/LocallyClosed.lean` where the API lemmas are? Thanks.

Comment (urkud) [reply Mathlib/Topology/Defs/Basic.lean:103]:
```suggestion
Also see `isLocallyClosed_tfae` and other lemmas in `Mathlib/Topology/LocallyClosed`.
```

File: Mathlib/Topology/Defs/Basic.lean [hunk 2]
```diff
@@ -110,6 +116,18 @@ def closure (s : Set X) : Set X :=
 def frontier (s : Set X) : Set X :=
   closure s \ interior s
 
+/--
+The coborder is defined as the complement of `closure s \ s`,
+or the union of `s` and the complement of `∂(s)`.
+This is the largest set such that `s` is closed in, and `s` is locally closed if and only if
```
Comment (grunweg) [inline Mathlib/Topology/Defs/Basic.lean:122]: I know you just moved the comment, but: I'm stumbling over the grammar of that sentence.
Do you mean "This is the largest set `t` such that `s` is closed in `t`", or "This is the largest set in which `s` is closed"?

**Review by grunweg — COMMENTED** [review]
Thanks for doing this! I have one pre-existing comment on a docstring.
maintainer merge

File: Mathlib/Topology/Defs/Basic.lean (hunk 2 above)
Comment (erdOne) [reply Mathlib/Topology/Defs/Basic.lean:122]: I added the comment in the first place. To me the the two clauses mean the same thing though. Can you please clarify?

Comment (j-loreaux) [reply Mathlib/Topology/Defs/Basic.lean:122]: I agree with Michael. As written, the sentence doesn't parse.
```suggestion
This is the largest set in which `s` is closed, and `s` is locally closed if and only if
```

**Comment by j-loreaux** [comment]
Forgive my ignorance, but is there an obvious reason the import chain should be reversed in the case of `LocallyClosed` and `LocalAtTarget`?

**Comment by urkud** [comment]
I think that `LocallyClosed` is a basic predicate like `IsOpen` or `IsClosed` while `LocalAtTarget` depends on `Topology.Sets.Opens` which depends on a lot of stuff.

**Comment by j-loreaux** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**